### PR TITLE
Feature/04 login using social media

### PR DESCRIPTION
**Pull request by peach0777**
### Changes Made

- Copy pasted css and images from provided html zip to the project's public directory. Each having its designated folder (folder name: css and img).
- Added 3 migration files 1) Added a new column "social_id" 2) Changed column "password" field to be nullable 3) Changed column "email" to be unique and nullable
- Grouped routes 
- Added a new controller "OAuthController" to handle login using socialite
- Added social media platform credentials

### CODE REVIEW

File: composer.lock
Comment (ChronoDevs) [inline composer.lock:1]: Note: Please don't commit this files is not part of our changes if possible
composer.lock
package.lock
vendor

Comment (peach0777) [reply composer.lock:1]: May I know the purpose of this deletion sir?

Comment (peach0777) [reply composer.lock:1]: Noted, sir. Will delete right away.

File: app/Http/Controllers/OAuthController.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Laravel\Socialite\Facades\Socialite;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+
+class OAuthController extends Controller
+{
+    public function redirect($socialPlatform) {
```
Comment (ChronoDevs) [inline app/Http/Controllers/OAuthController.php]: can you add comment every function like this

```
/**
 * Redirect QAuth login
 * 
 * @param string $socialPlatform
 * @return response
 */
public function redirect($socialPlatform)
{
}
```

Comment (peach0777) [reply app/Http/Controllers/OAuthController.php]: Noted, sir.

File: app/Http/Controllers/OAuthController.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return Socialite::driver($socialPlatform)->redirect();
+    }
+
+    public function loginUsingSocial($socialPlatform) {
+        $socialUser = Socialite::driver($socialPlatform)->user();
```
Comment (ChronoDevs) [inline app/Http/Controllers/OAuthController.php:31]: show error message if don't have provider_id response

Comment (peach0777) [reply app/Http/Controllers/OAuthController.php:31]: Noted, sir.

File: composer.lock
Comment (ChronoDevs) [reply composer.lock:1]: Please also commit .env.example if you have added credentials on your .env. Thank you

File: app/Http/Controllers/OAuthController.php (hunk 2 above)
Comment (peach0777) [reply app/Http/Controllers/OAuthController.php:31]: By show message, do you mean show message in the blade/view or just in console sir?

Comment (ChronoDevs) [reply app/Http/Controllers/OAuthController.php:31]: error message in the .blade

File: app/Http/Controllers/OAuthController.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     * Updates/Creates user using social info and log in
+     * 
+     * @param string $socialPlatform
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function loginUsingSocial($socialPlatform)
+    {
+        $socialUser = Socialite::driver($socialPlatform)->user();
+
+        if(!isset($socialUser->id)) {
```
Comment (ChronoDevs) [inline app/Http/Controllers/OAuthController.php]: add space for if statement
if(!isset($socialUser->id))  => if (!isset($socialUser->id))

Comment (peach0777) [reply app/Http/Controllers/OAuthController.php]: Noted, sir.

**Pull request merged**